File: Stack/Evaluate Reverse Polish Notation/Evaluate.cs
class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> st = new Stack<int>();
        
        foreach (string token in tokens) {
            if (IsOperator(token)) {
                int operand2 = st.Pop();
                int operand1 = st.Pop();
                st.Push(PerformOperation(operand1, operand2, token));
            } else {
                st.Push(int.Parse(token));
            }
        }
        
        return st.Pop();
    }

    private bool IsOperator(string token) {
        return (token == "+" || token == "-" || token == "*" || token == "/");
    }

    private int PerformOperation(int operand1, int operand2, string op) {
        switch (op) {
            case "+":
                return operand1 + operand2;
            case "-":
                return operand1 - operand2;
            case "*":
                return operand1 * operand2;
            case "/":
                return operand1 / operand2;
            default:
                return 0;
        }
    }
}

File: Stack/Evaluate Reverse Polish Notation/Evaluate.cpp
class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        stack<int> st;
        
        for (const string& token : tokens) {
            if (isOperator(token)) {
                int operand2 = st.top(); st.pop();
                int operand1 = st.top(); st.pop();
                st.push(performOperation(operand1, operand2, token));
            } else {
                st.push(stoi(token));
            }
        }
        
        return st.top();
    }

private:
    bool isOperator(const string& token) {
        return (token == "+" || token == "-" || token == "*" || token == "/");
    }

    int performOperation(int operand1, int operand2, const string& op) {
        switch (op[0]) {
            case '+':
                return operand1 + operand2;
            case '-':
                return operand1 - operand2;
            case '*':
                return operand1 * operand2;
            case '/':
                return operand1 / operand2;
            default:
                return 0;
        }
    }
};